File: 1109-corporate-flight-bookings/1109. Corporate Flight Bookings.py
class Solution:
    def corpFlightBookings(self, bookings: List[List[int]], n: int) -> List[int]:
        res = [0] * (n+1)
        for start, end, val in bookings:
            res[start-1] += val
            res[end] -= val
        res = list(accumulate(res))
        return res[:-1]

File: 1109-corporate-flight-bookings/1109. Corporate Flight Bookings.cpp
class Solution {
public:
    vector<int> corpFlightBookings(vector<vector<int>>& bookings, int n) {
        vector<int> r(n+1, 0);
        int a, b, c;
        for (int i = 0; i < bookings.size(); i++) {
            a = bookings[i][0];
            b = bookings[i][1];
            c = bookings[i][2];
            r[a - 1] += c;
            r[b] -= c;
        }
        cout << r[1] << endl;
        int curr = 0;
        for (int i = 0; i < n; i++) {
            curr += r[i];
            r[i] = curr;
        }
        r.pop_back();
        return r;
    }
};